irida_staramr_results/irida_api.py

```diff
@@ -112,112 +110,169 @@ def _create_session(self):
+                logging.error(error_txt)
+                raise exceptions.IridaKeyError(error_txt)
+
+            # try to get all keys from target_dict to our list or links
+            try:
+                links_list = next(
+                    r["links"] for r in resources_list
+                    if r[target_dict["key"]].lower() == str(target_dict["value"]).lower()
+                )
+
+            except KeyError:
+                raise exceptions.IridaKeyError(target_dict["key"] + " not found. Available keys: "
+                                                                    ", ".join(resources_list[0].keys()))
 
-    def get_project_from_id(self, _id):
-        """ Returns a project resource object from an id of a project """
-        return self._get_resource_from_path(f"/projects/{_id}")
+            except StopIteration:
+                raise exceptions.IridaKeyError(target_dict["value"] + " not found.")
 
-    def get_analyses_from_projects(self, _id):
+
+        else:
+            links_list = response.json()["resource"]["links"]
+
+        # find the right "href" based on "rel"/target_key
+        try:
+            ret_val = next(link["href"] for link in links_list
+                           if link["rel"] == target_key)
+        except StopIteration:
+            error_txt = target_key + " not found in links. Available links: " +\
+                        ", ".join([str(link["rel"]) for link in links_list])
+            logging.debug(error_txt)
+            raise exceptions.IridaKeyError(error_txt)
+
+        return ret_val
+
+    def get_amr_analysis_results(self, project_id):
+        """
+        Get all AMR detection analysis results from a project id.
+        If nothing is found, it returns an empty array.
+        :param project_id:
+        :return analysis_result_list:
         """
-        Returns an array containing analysis objects for a given project
 
-        :param _id:
+        try:
+            project_analysis_submissions = self._get_analysis_submissions_from_projects(project_id)
+        except KeyError:
+            error_txt = "No analysis found in project."
+            logging.error(error_txt)
+            raise exceptions.IridaKeyError(error_txt)
+
+        all_analysis_results = self._get_all_analysis_results(project_analysis_submissions)
+
+        # Filter AMR Detection results
+        amr_analysis_results = [analysis_result for analysis_result in all_analysis_results if
+                                self._is_type_amr(analysis_result)]
+
+        return amr_analysis_results
+
+    def _get_analysis_submissions_from_projects(self, project_id):
+        """
+        Returns an array of all analyses for a given project
+        :param project_id:
         :return project_analysis_list:
         """
-        project_analysis_list = []
+        try:
+            project_url = self._get_link(self.base_url + f"/api/projects/{project_id}", "self")
```

Comment:
A few notes on using `_get_link`
* We should assume that the base_url already includes the `/api/` as part of the base, so don't include it when getting the link, but instead include it when initializing the api
* When adhering to HATEOAS it's better to query in smaller pieces s.t. we fetch projects, then use the target_dict to get the analyses, instead of assuming the url exists where it currently exists. LIke this:
```
project_url = self._get_link(self.base_url, "projects")
analysis_submissions_url = self._get_link(project_url, "project/analyses",
                            target_dict={
                                "key": "identifier",
                                "value": project_id
                            })
```
This should add safety for fetching a project when its possible a user does not have permission to access it.
(I'm not sure the exact code above would work, or if the target dict has analyses links on it, please let me know if/how this works)